scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala

```diff
@@ -306,67 +336,81 @@ class BigQueryClient private (private val projectId: String,
       rows, schema, writeDisposition, createDisposition)
 
   /** Wait for all jobs to finish. */
-  def waitForJobs(jobs: QueryJob*): Unit = {
+  def waitForJobs(jobs: BigQueryJob*): Unit = {
     val numTotal = jobs.size
-    var pendingJobs = jobs.filter(_.jobReference.isDefined)
+    var pendingJobs = jobs.flatMap {
+      job =>
+        job.jobReference match {
+          case Some(reference) => Some(job, reference)
+          case None => None
+        }
+    }
+
     while (pendingJobs.nonEmpty) {
-      val remainingJobs = pendingJobs.filter { j =>
-        val jobId = j.jobReference.get.getJobId
-        try {
-          val poll = bigquery.jobs().get(projectId, jobId).execute()
-          val error = poll.getStatus.getErrorResult
-          if (error != null) {
-            throw new RuntimeException(s"Query job failed: id: $jobId, error: $error")
-          }
-          if (poll.getStatus.getState == "DONE") {
-            logJobStatistics(j.query, poll)
-            false
-          } else {
-            true
+      val remainingJobs = pendingJobs.filter {
+        case (bqJob, jobReference) =>
+          val jobId = jobReference.getJobId
+          try {
+            val poll = bigquery.jobs().get(projectId, jobId).execute()
+            val error = poll.getStatus.getErrorResult
+            if (error != null) {
+              throw new RuntimeException(s"${bqJob.jobType} Job failed: id: $jobId, error: $error")
+            }
+            if (poll.getStatus.getState == "DONE") {
+              logJobStatistics(bqJob, poll)
+              false
+            } else {
+              true
+            }
+          } catch {
+            case e: IOException =>
+              logger.warn(s"BigQuery request failed: id: $jobId, error: $e")
+              true
           }
-        } catch {
-          case e: IOException =>
-            logger.warn(s"BigQuery request failed: id: $jobId, error: $e")
-            true
-        }
       }
 
       pendingJobs = remainingJobs
       val numDone = numTotal - pendingJobs.size
-      logger.info(s"Query: $numDone out of $numTotal completed")
+      logger.info(s"Job: $numDone out of $numTotal completed")
       if (pendingJobs.nonEmpty) {
         Thread.sleep(10000)
       }
     }
   }
 
+  /** Delete table */
+  private[bigquery] def deleteTable(table: TableReference): Unit = {
+    bigquery.tables().delete(table.getProjectId, table.getDatasetId, table.getTableId)
+      .execute()
+  }
+
   // =======================================================================
   // Type safe API
   // =======================================================================
 
   /**
-   * Get a typed iterator for a BigQuery SELECT query or table.
```

Comment:
Haven't looked thoroughly yet but you should fix scaladoc style in Intellij to avoid changing these:
https://github.com/spotify/scio/wiki/Style-Guide#documentation-style